python/cudf/cudf/core/column/categorical.py

```diff
@@ -1095,18 +1094,21 @@ def find_and_replace(
         # map it to the new label it is to be replaced by
         dtype_replace = cudf.Series(replacement_col)
         dtype_replace[dtype_replace.isin(old_cats["cats"])] = None
-        new_cats["cats"] = new_cats["cats"].replace(
+        old_cats._data["new_cats"] = old_cats["cats"].replace(
```

Comment:
It's awkward to have a column `"new_cats"` in a dataframe with the variable name `old_cats`. Would it make more sense to leave this in a variable `new_cats_col`, rather than add a column in the `old_cats` dataframe named `new_cats`? I see it is used immediately below to construct the DataFrame `new_cats`. I don't immediately see a good way to resolve all of the awkward name reuse.

Reply:
Changed it 👍